plugin/pkg/shoot/validator/admission.go

```diff
@@ -2197,6 +2201,51 @@ func validateMaxNodesTotal(workers []core.Worker, maxNodesTotal int32) field.Err
 	return allErrs
 }
 
+func (c *validationContext) validateDefaultDomainCompatibilityForRescheduling(oldSeed *gardencorev1beta1.Seed) error {
+	if c.shoot.Spec.DNS == nil || c.shoot.Spec.DNS.Domain == nil {
+		return nil
+	}
+
+	shootDomain := *c.shoot.Spec.DNS.Domain
+
+	oldDefaultDomains := getDefaultDomainsForSeed(oldSeed)
+
+	var (
+		usesDefaultDomain bool
+		usedDefaultDomain string
+	)
+
+	for _, defaultDomain := range oldDefaultDomains {
+		if strings.HasSuffix(shootDomain, "."+defaultDomain) {
+			usesDefaultDomain = true
+			usedDefaultDomain = defaultDomain
+			break
+		}
+	}
+
+	if !usesDefaultDomain {
+		return nil
+	}
+
+	newDefaultDomains := getDefaultDomainsForSeed(c.seed)
+
+	if slices.Contains(newDefaultDomains, usedDefaultDomain) {
+		return nil
+	}
+
+	return fmt.Errorf("cannot reschedule shoot %q to seed %q because the shoot uses default domain %q which is not supported by the new seed (supported domains: %v)", c.shoot.Name, c.seed.Name, usedDefaultDomain, newDefaultDomains)
```

Comment:
```suggestion
	return fmt.Errorf("cannot reschedule shoot %q to seed %q because the shoot uses the default domain %q which is not supported by the new seed (supported domains: %v)", c.shoot.Name, c.seed.Name, usedDefaultDomain, newDefaultDomains)
```